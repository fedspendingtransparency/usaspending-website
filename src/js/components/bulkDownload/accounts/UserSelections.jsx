/**
 * UserSelections.jsx
 * Created by Lizzie Salita 4/25/18
 */

import React from 'react';
import PropTypes from 'prop-types';

import { accountDownloadOptions } from 'dataMapping/bulkDownload/bulkDownloadOptions';

const propTypes = {
    accounts: PropTypes.object
};

export default class UserSelections extends React.Component {
    constructor(props) {
        super(props);

        this.generateAccountLevelString = this.generateAccountLevelString.bind(this);
        this.generateAgencyString = this.generateAgencyString.bind(this);
        this.generateBudgetFunctionString = this.generateBudgetFunctionString.bind(this);
        this.generateFederalAccountString = this.generateFederalAccountString.bind(this);
        this.generateSubmissionTypeString = this.generateSubmissionTypeString.bind(this);
        this.generateFyString = this.generateFyString.bind(this);
    }

    generateAccountLevelString() {
        if (this.props.accounts.accountLevel) {
            const options = accountDownloadOptions.accountLevels;
            const selectedOption = options.find((option) =>
                option.name === this.props.accounts.accountLevel
            );
            return (
                <div className="selection__content">{selectedOption.label}</div>
            );
        }
        return (
            <div className="selection__content selection__content-required">Account not selected</div>
        );
    }

    generateAgencyString() {
        if (this.props.accounts.agency.name !== 'Select an Agency') {
            return (
                <div className="selection__content">{this.props.accounts.agency.name}</div>
            );
        }

        return (
            <div className="selection__content selection__content-required">Budget function or Agency required</div>
        );
    }

    generateBudgetFunctionString() {
        if (this.props.accounts.budgetFunction.title !== 'Select a Budget Function') {
            return (
                <div className="selection__content">{this.props.accounts.budgetFunction.title}</div>
            );
        }

        return (
            <div className="selection__content selection__content-required">Budget function or Agency required</div>
        );
    }

    generateFederalAccountString() {
        if (this.props.accounts.federalAccount.name !== 'Select a Federal Account') {
            return (
                <div className="selection__content">{this.props.accounts.federalAccount.name}</div>
            );
        }

        return (
            <div className="selection__content">Federal account not selected</div>
        );
    }

    generateBudgetSubfunctionString() {
        if (this.props.accounts.budgetSubfunction.title !== 'Select a Budget Sub-Function') {
            return (
                <div className="selection__content">{this.props.accounts.budgetSubfunction.title}</div>
            );
        }

        return (
            <div className="selection__content">Budget sub-function not selected</div>
        );
    }

    generateSubmissionTypeString() {
        if (this.props.accounts.submissionTypes.length > 0) {
            return (
                <div className="selection__content">
                    {accountDownloadOptions.submissionTypes
                        .filter((option) => this.props.accounts.submissionTypes.includes(option.name))
                        .reduce((acc, option, i, array) => {
                            // don't append comma
                            if (i === 0 && array.length === 1) return `${option.label}`;
                            if (i === array.length - 1) return `${acc}${option.label}`;
                            return `${acc}${option.label}, `;
                        }, '')}
                </div>
            );
        }
        return (
            <div className="selection__content selection__content-required">required</div>
        );
    }

    generateFyString() {
        if (this.props.accounts.fy && this.props.accounts.quarter) {
            return (
                <div className="selection__content">{this.props.accounts.fy} - Q{this.props.accounts.quarter}</div>
            );
        }

        return (
            <div className="selection__content selection__content-required">required</div>
        );
    }

    render() {
        return (
            <div className="download-user-selections">
                <h3 className="download-user-selections__title">Your selected options are...</h3>
                <div className="download-user-selections__left-col">
                    <div className="selection">
                        <div className="selection__heading">Budget Function</div>
                        <div className="selection__content">
                            {this.generateBudgetFunctionString()}
                        </div>
                    </div>
                    <div className="selection">
                        <div className="selection__heading">Budget Sub-Function</div>
                        <div className="selection__content">
                            {this.generateBudgetSubfunctionString()}
                        </div>
                    </div>
                    <div className="selection">
                        <div className="selection__heading">Agency</div>
                        {this.generateAgencyString()}
                    </div>
                    <div className="selection">
                        <div className="selection__heading">Federal Account</div>
                        <div className="selection__content">
                            {this.generateFederalAccountString()}
                        </div>
                    </div>

                    <div className="selection">
                        <div className="selection__heading">Account Level</div>
                        <div className="selection__content">
                            {this.generateAccountLevelString()}
                        </div>
                    </div>
                </div>
                <div className="download-user-selections__right-col">
                    <div className="selection">
                        <div className="selection__heading">File Submission Type</div>
                        {this.generateSubmissionTypeString()}
                    </div>
                    <div className="selection">
                        <div className="selection__heading">Fiscal Year</div>
                        {this.generateFyString()}
                    </div>
                </div>
            </div>
        );
    }
}

UserSelections.propTypes = propTypes;
