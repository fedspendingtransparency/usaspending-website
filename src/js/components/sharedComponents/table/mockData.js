
export const mockRows = [
    {
        award_id: "0001",
        subaward_count: 1510,
        award_generated_internal_id: "CONT_AWD_0001_9700_FA870215D0001_9700",
        subaward_obligation: 409303652.12
    },
    {
        award_id: "N0001923C0003",
        subaward_count: 1723,
        award_generated_internal_id: "CONT_AWD_N0001923C0003_9700_-NONE-_-NONE-",
        subaward_obligation: 2406096453.7
    },
    {
        award_id: "N0002416C2229",
        subaward_count: 1871,
        award_generated_internal_id: "CONT_AWD_N0002416C2229_9700_-NONE-_-NONE-",
        subaward_obligation: 552408209.26
    },
    {
        award_id: "N0002415C4301",
        subaward_count: 1974,
        award_generated_internal_id: "CONT_AWD_N0002415C4301_9700_-NONE-_-NONE-",
        subaward_obligation: 658930044.09
    },
    {
        award_id: "N0001921C0020",
        subaward_count: 2545,
        award_generated_internal_id: "CONT_AWD_N0001921C0020_9700_-NONE-_-NONE-",
        subaward_obligation: 492379269.3
    },
    {
        award_id: "W56HZV15C0095",
        subaward_count: 2676,
        award_generated_internal_id: "CONT_AWD_W56HZV15C0095_9700_-NONE-_-NONE-",
        subaward_obligation: 304424719.1
    },
    {
        award_id: "DENA0001942",
        subaward_count: 2976,
        award_generated_internal_id: "CONT_AWD_DENA0001942_8900_-NONE-_-NONE-",
        subaward_obligation: 1572986547.25
    },
    {
        award_id: "N0001920C0009",
        subaward_count: 4345,
        award_generated_internal_id: "CONT_AWD_N0001920C0009_9700_-NONE-_-NONE-",
        subaward_obligation: 419902669.32
    },
    {
        award_id: "N0002418C2106",
        subaward_count: 12407,
        award_generated_internal_id: "CONT_AWD_N0002418C2106_9700_-NONE-_-NONE-",
        subaward_obligation: 918283598.01
    },
    {
        award_id: "N0002416C2116",
        subaward_count: 45608,
        award_generated_internal_id: "CONT_AWD_N0002416C2116_9700_-NONE-_-NONE-",
        subaward_obligation: 2666421624.76
    }
];
