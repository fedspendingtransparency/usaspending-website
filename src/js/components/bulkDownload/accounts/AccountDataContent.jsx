/**
 * AccountDataContent.jsx
 * Created by Lizzie Salita 4/23/18
 */

import React from 'react';
import PropTypes from 'prop-types';
import { isEqual } from 'lodash';
import kGlobalConstants from 'GlobalConstants';

import { accountDownloadOptions } from 'dataMapping/bulkDownload/bulkDownloadOptions';
import { Glossary } from 'components/sharedComponents/icons/Icons';

import AccountLevelFilter from './filters/AccountLevelFilter';
import AgencyFilter from './filters/AgencyFilter';
import BudgetFunctionFilter from './filters/BudgetFunctionFilter';
import SubmissionTypeFilter from './filters/SubmissionTypeFilter';
import FiscalYearFilter from './filters/FiscalYearFilter';
import UserSelections from './UserSelections';
import SubmitButton from '../awards/SubmitButton';

import FilterSelection from './filters/FilterSelection';

const propTypes = {
    accounts: PropTypes.object,
    updateFilter: PropTypes.func,
    clearAccountFilters: PropTypes.func,
    agencies: PropTypes.object,
    federalAccounts: PropTypes.array,
    clickedDownload: PropTypes.func,
    setFederalAccountList: PropTypes.func,
    budgetFunctions: PropTypes.array,
    setBudgetSubfunctionList: PropTypes.func,
    budgetSubfunctions: PropTypes.array
};

export default class AccountDataContent extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            validForm: false
        };

        this.handleSubmit = this.handleSubmit.bind(this);
        this.resetForm = this.resetForm.bind(this);
    }

    componentDidUpdate(prevProps) {
        if (!isEqual(prevProps.accounts, this.props.accounts)) {
            this.validateForm(this.props.accounts);
        }
    }

    componentWillUnmount() {
        this.props.clearAccountFilters();
    }

    handleSubmit() {
        this.props.clickedDownload();
    }

    resetForm() {
        this.props.clearAccountFilters();
    }

    validateForm(accounts) {
        const validForm = (
            (accounts.budgetFunction.code !== '')
            && (accounts.agency.id !== '')
            && (accounts.submissionTypes.length !== 0)
            && (accounts.fy !== '')
            && (accounts.quarter !== '' || accounts.period !== '')
        );

        this.setState({
            validForm
        });
    }

    render() {
        const accounts = this.props.accounts;
        return (
            <div className="download-center">
                <div className="download-center__filters">
                    <h2 className="download-center__title">Custom Account Data</h2>
                    <FilterSelection valid={accounts.budgetFunction.code !== '' || accounts.agency.id !== ''} />
                    <form className="download-center-form">
                        <BudgetFunctionFilter
                            budgetFunctions={this.props.budgetFunctions}
                            budgetSubfunctions={this.props.budgetSubfunctions}
                            currentBudgetFunction={accounts.budgetFunction}
                            currentBudgetSubfunction={accounts.budgetSubfunction}
                            setBudgetSubfunctionList={this.props.setBudgetSubfunctionList}
                            updateFilter={this.props.updateFilter}
                            validAgencyId={accounts.agency.id !== ''}
                            valid={accounts.budgetFunction.code !== ''} />
                        <AgencyFilter
                            agencies={this.props.agencies}
                            federalAccounts={this.props.federalAccounts}
                            currentAgency={accounts.agency}
                            currentFederalAccount={accounts.federalAccount}
                            setFederalAccountList={this.props.setFederalAccountList}
                            updateFilter={this.props.updateFilter}
                            validBudgetFunctionCode={accounts.budgetFunction.code !== ''}
                            valid={accounts.agency.id !== ''} />
                        <AccountLevelFilter
                            accountLevels={accountDownloadOptions.accountLevels}
                            currentAccountLevel={accounts.accountLevel}
                            updateFilter={this.props.updateFilter}
                            valid={accounts.accountLevel !== ''} />
                        <SubmissionTypeFilter
                            submissionTypes={accountDownloadOptions.submissionTypes}
                            currentSubmissionTypes={accounts.submissionTypes}
                            updateFilter={this.props.updateFilter}
                            valid={accounts.submissionTypes.length !== 0} />
                        <FiscalYearFilter
                            currentFy={accounts.fy}
                            latestSelectedTimePeriod={accounts.period === '' ? accounts.quarter : accounts.period}
                            updateFilter={this.props.updateFilter}
                            valid={(accounts.fy && (accounts.quarter || accounts.period))} />
                        <UserSelections
                            accounts={accounts} />
                        <SubmitButton
                            handleSubmit={this.handleSubmit}
                            validForm={this.state.validForm}
                            filters={accounts}
                            validDates
                            dataType="accounts" />
                    </form>
                    <button className="download-center__reset" onClick={this.resetForm}>
                        Reset form and start over
                    </button>
                </div>
                <div className="download-info">
                    <h3 className="download-info__title">About Account Data</h3>
                    <div className="download-info__section">
                        <h4 className="download-info__section-heading">What is account data?</h4>
                        <p>
                            Account data covers all spending data, including non-award spending.
                        </p>
                        <p>
                            The data is available on two different levels, <strong>federal account</strong>&nbsp;
                            <a href="#/download_center/custom_account_data/?glossary=federal-account"><Glossary /></a>
                            and <strong>treasury account</strong>&nbsp;
                            <a href="#/download_center/custom_account_data/?glossary=treasury-account-symbol-tas"><Glossary /></a>
                            . Federal account data is essentially a &ldquo;roll-up&rdquo; of multiple treasury account data.
                        </p>
                        <p>
                            The files available are categorized by type, according to the scope of spending they cover. More information on the different file types can be found in our <a href={`https://files${kGlobalConstants.DEV ? '-nonprod' : ''}.usaspending.gov/docs/Custom+Account+Data+Dictionary.xlsx`}>Custom Account Data Dictionary</a>.
                        </p>
                    </div>
                    <div className="download-info__section">
                        <h4 className="download-info__section-heading">Why is this data useful?</h4>
                        <p>
                            Account data contains the most encompassing amounts of spending throughout U.S. government agencies.  Unlike award data, account data include spending that is not tied to awards, such as operational costs and employee salaries.
                        </p>
                    </div>
                    <div className="download-info__section">
                        <h4 className="download-info__section-heading">How do I use this form?</h4>
                        <p>
                            This form allows you to download account data in a range of quarters within a specific fiscal year.
                        </p>
                        <p>
                            Select an option in each section and click the &ldquo;Download&rdquo; button at the bottom.
                        </p>
                        <p>
                            Heads up: all fields are required. You&rsquo;ll only be able to start the download when all sections are properly filled.
                        </p>
                    </div>
                </div>
            </div>
        );
    }
}

AccountDataContent.propTypes = propTypes;
