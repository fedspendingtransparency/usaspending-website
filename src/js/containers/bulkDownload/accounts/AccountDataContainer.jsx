/**
 * AccountDataContainer.jsx
 * Created by Lizzie Salita 4/23/18
 */

import React from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import * as BulkDownloadHelper from 'helpers/bulkDownloadHelper';
import * as bulkDownloadActions from 'redux/actions/bulkDownload/bulkDownloadActions';

import AccountDataContent from 'components/bulkDownload/accounts/AccountDataContent';

const propTypes = {
    updateDownloadFilter: PropTypes.func,
    clearDownloadFilters: PropTypes.func,
    bulkDownload: PropTypes.object,
    clickedDownload: PropTypes.func
};

export class AccountDataContainer extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            agencies: {
                cfoAgencies: [],
                otherAgencies: []
            },
            federalAccounts: [],
            budgetFunctions: [],
            budgetSubfunctions: []
        };

        this.agencyListRequest = null;
        this.federalAccountListRequest = null;

        this.budgetFunctionListRequest = null;
        this.budgetSubfunctionListRequest = null;

        this.updateFilter = this.updateFilter.bind(this);
        this.clearAccountFilters = this.clearAccountFilters.bind(this);
        this.setAgencyList = this.setAgencyList.bind(this);
        this.setFederalAccountList = this.setFederalAccountList.bind(this);
        this.setBudgetFunctionList = this.setBudgetFunctionList.bind(this);
        this.setBudgetSubfunctionList = this.setBudgetSubfunctionList.bind(this);
    }

    componentDidMount() {
        this.setAgencyList();
        this.setBudgetFunctionList();
    }

    setAgencyList() {
        if (this.agencyListRequest) {
            this.agencyListRequest.cancel();
        }

        // perform the API request
        this.agencyListRequest = BulkDownloadHelper.requestAgenciesList({
            type: "account_agencies",
            agency: 0
        });

        this.agencyListRequest.promise
            .then((res) => {
                const cfoAgencies = res.data.agencies.cfo_agencies;
                const otherAgencies = res.data.agencies.other_agencies;
                this.setState({
                    agencies: {
                        cfoAgencies,
                        otherAgencies
                    }
                });
            })
            .catch((err) => {
                console.log(err);
                this.agencyListRequest = null;
            });
    }

    async setFederalAccountList(agencyCode, page = 1) {
        if (agencyCode !== '') {
            if (this.federalAccountListRequest) {
                this.federalAccountListRequest.cancel();
            }

            this.federalAccountListRequest = BulkDownloadHelper.requestFederalAccountList(agencyCode, page);
            try {
                const { data } = await this.federalAccountListRequest.promise;
                this.setState({
                    federalAccounts: page > 1
                        // we're requesting the second page, concat array
                        ? [...this.state.federalAccounts, ...data.results]
                        : data.results
                });
                if (data.hasNext) {
                    this.setFederalAccountList(agencyCode, page + 1);
                }
            }
            catch (e) {
                console.log(e);
                this.federalAccountListRequest = null;
            }
        }
        else {
            this.setState({
                federalAccounts: []
            }, () => {
                this.resetFederalAccount();
            });
        }
    }

    setBudgetFunctionList() {
        if (this.budgetFunctionListRequest) {
            this.budgetFunctionListRequest.cancel();
        }

        // perform the API request
        this.budgetFunctionListRequest = BulkDownloadHelper.requestBudgetFunctionList();

        this.budgetFunctionListRequest.promise
            .then((res) => {
                const budgetFunctions = res.data.results;
                this.setState({
                    budgetFunctions
                });
            })
            .catch((err) => {
                console.log(err);
                this.budgetFunctionListRequest = null;
            });
    }

    setBudgetSubfunctionList(budgetFunction) {
        if (budgetFunction !== '') {
            if (this.budgetSubfunctionListRequest) {
                this.budgetSubfunctionListRequest.cancel();
            }

            this.budgetSubfunctionListRequest = BulkDownloadHelper.requestBudgetSubfunctionList({
                budget_function: budgetFunction
            });

            this.budgetSubfunctionListRequest.promise
                .then((res) => {
                    const budgetSubfunctions = res.data.results;
                    this.setState({
                        budgetSubfunctions
                    });
                })
                .catch((err) => {
                    console.log(err);
                    this.budgetSubfunctionListRequest = null;
                });
        }
        else {
            this.setState({
                budgetSubfunctions: []
            }, () => {
                this.resetBudgetSubfunction();
            });
        }
    }

    updateFilter(name, value) {
        this.props.updateDownloadFilter({
            dataType: 'accounts',
            name,
            value
        });
    }

    resetFederalAccount() {
        this.updateFilter('federalAccount', {
            id: '',
            name: 'Select a Federal Account'
        });
    }


    clearAccountFilters() {
        this.props.clearDownloadFilters('accounts');
    }

    resetBudgetSubfunction() {
        this.updateFilter('budgetSubfunction', {
            code: '',
            title: 'Select a Budget Sub-Function'
        });
    }

    render() {
        return (
            <AccountDataContent
                accounts={this.props.bulkDownload.accounts}
                federalAccounts={this.state.federalAccounts}
                setFederalAccountList={this.setFederalAccountList}
                updateFilter={this.updateFilter}
                clearAccountFilters={this.clearAccountFilters}
                agencies={this.state.agencies}
                budgetFunctions={this.state.budgetFunctions}
                budgetSubfunctions={this.state.budgetSubfunctions}
                setBudgetSubfunctionList={this.setBudgetSubfunctionList}
                clickedDownload={this.props.clickedDownload} />
        );
    }
}

AccountDataContainer.propTypes = propTypes;

export default connect(
    (state) => ({ bulkDownload: state.bulkDownload }),
    (dispatch) => bindActionCreators(bulkDownloadActions, dispatch)
)(AccountDataContainer);
