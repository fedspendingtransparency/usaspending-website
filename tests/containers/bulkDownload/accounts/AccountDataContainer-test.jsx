/**
 * AccountDataContainer-test.jsx
 * Created by Lizzie Salita 4/25/18
 */

import React from 'react';
import { mount, shallow } from 'enzyme';
import { AccountDataContainer } from 'containers/bulkDownload/accounts/AccountDataContainer';
import { mockActions, mockAgencies, mockFederalAccounts, mockBudgetFunctions, mockProps } from '../mockData';

// mock the bulkDownload helper
jest.mock('helpers/bulkDownloadHelper', () => require('../mockBulkDownloadHelper'));

// mock the child component by replacing it with a function that returns a null element
jest.mock('components/bulkDownload/accounts/AccountDataContent', () => jest.fn(() => null));

describe('AccountDataContainer', () => {
    it('should make an API call for the agencies on mount', async () => {
        const container = mount(<AccountDataContainer
            {...mockActions}
            bulkDownload={mockProps} />);

        const expectedState = {
            cfoAgencies: mockAgencies.cfo_agencies,
            otherAgencies: mockAgencies.other_agencies
        };

        await container.instance().agencyListRequest.promise;
        expect(container.state().agencies).toEqual(expectedState);
    });

    describe('setFederalAccountList', () => {
        it('should make an API request when called and update state', async () => {
            const container = shallow(<AccountDataContainer
                {...mockActions}
                bulkDownload={mockProps} />);

            const expectedState = mockFederalAccounts;
            container.instance().setFederalAccountList('02');
            await container.instance().federalAccountListRequest.promise;
            expect(container.state().federalAccounts).toEqual(expectedState);
        });

        it('should append the results to the existing result state if the page number is greater than 1', async () => {
            const container = shallow(<AccountDataContainer
                {...mockActions}
                bulkDownload={mockProps} />);

            container.setState({
                federalAccounts: [{}, {}, {}]
            });

            expect(container.state().federalAccounts.length).toEqual(3);
            await container.instance().setFederalAccountList('02', 2);
            expect(container.state().federalAccounts.length).toEqual(5);
        });
    });
    it('should make an API call for the budget functions on mount', async () => {
        const container = mount(<AccountDataContainer
            {...mockActions}
            bulkDownload={mockProps} />);

        const expectedState = mockBudgetFunctions;

        await container.instance().budgetFunctionListRequest.promise;
        expect(container.state().budgetFunctions).toEqual(expectedState);
    });
});
