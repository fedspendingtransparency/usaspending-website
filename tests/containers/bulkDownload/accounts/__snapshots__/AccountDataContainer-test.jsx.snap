// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountDataContainer tests populates the agency list 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "accounts": {
          "accountLevel": "treasuryAccount",
          "agency": {
            "id": "123",
            "name": "Mock Agency",
          },
          "budgetFunction": {
            "code": "300",
            "title": "Mock Budget Function",
          },
          "budgetSubfunction": {
            "code": "123",
            "title": "Mock Budget Sub Function",
          },
          "defCodes": [
            "L",
            "M",
            "N",
            "O",
            "P",
          ],
          "federalAccount": {
            "id": "212",
            "name": "Mock Federal Account",
          },
          "fileFormat": "csv",
          "fy": "1989",
          "period": "",
          "quarter": "1",
          "submissionTypes": [
            "accountBalances",
          ],
        },
        "agencies": {
          "cfoAgencies": [],
          "otherAgencies": [],
        },
        "budgetFunctions": [],
        "budgetSubfunctions": [],
        "clearAccountFilters": [Function],
        "clickedDownload": [MockFunction],
        "federalAccounts": [],
        "setBudgetSubfunctionList": [Function],
        "setDefCodes": [MockFunction],
        "setFederalAccountList": [Function],
        "updateFilter": [Function],
      },
      undefined,
    ],
    [
      {
        "accounts": {
          "accountLevel": "treasuryAccount",
          "agency": {
            "id": "123",
            "name": "Mock Agency",
          },
          "budgetFunction": {
            "code": "300",
            "title": "Mock Budget Function",
          },
          "budgetSubfunction": {
            "code": "123",
            "title": "Mock Budget Sub Function",
          },
          "defCodes": [
            "L",
            "M",
            "N",
            "O",
            "P",
          ],
          "federalAccount": {
            "id": "212",
            "name": "Mock Federal Account",
          },
          "fileFormat": "csv",
          "fy": "1989",
          "period": "",
          "quarter": "1",
          "submissionTypes": [
            "accountBalances",
          ],
        },
        "agencies": {
          "cfoAgencies": "stuff",
          "otherAgencies": "other stuff",
        },
        "budgetFunctions": [
          {
            "budget_function_code": "300",
            "budget_function_title": "Mock Budget Function",
          },
        ],
        "budgetSubfunctions": [],
        "clearAccountFilters": [Function],
        "clickedDownload": [MockFunction],
        "federalAccounts": [],
        "setBudgetSubfunctionList": [Function],
        "setDefCodes": [MockFunction],
        "setFederalAccountList": [Function],
        "updateFilter": [Function],
      },
      undefined,
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": null,
    },
    {
      "type": "return",
      "value": null,
    },
  ],
}
`;
